packages/client/components/TeamPrompt/Recurrence/RecurrenceSettings.tsx

```diff
@@ -0,0 +1,217 @@
+  padding: 16,
+  fontSize: 14
+})
+
+const RecurrenceSettingsTitle = styled('div')({
+  fontSize: 18,
+  fontWeight: 600
+})
+
+const HumanReadableRecurrenceRule = styled('div')({
+  fontSize: 14,
+  maxWidth: 360,
+  overflow: 'hidden'
+})
+
+const RecurrenceDayPickerRoot = styled('div')({
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+  gap: 8,
+  marginTop: 16,
+  marginBottom: 8
+})
+
+const StartTimeDropdownToggle = styled(DropdownMenuToggle)({
+  fontSize: 14,
+  width: '100%',
+  marginTop: 8,
+  '&:hover, :focus, :focus-visible, :active': {
+    outline: `1px solid ${PALETTE.SLATE_600}`,
+    borderColor: PALETTE.SLATE_600,
+    borderRadius: 4
+  }
+})
+
+const StartTimeSection = styled('div')({
+  marginTop: 16
+})
+
+interface Props {
+  parentId: PortalId
+  onRecurrenceRuleUpdated: (rrule: RRule | null) => void
+  recurrenceRule: RRule | null
+}
+
+export const RecurrenceSettings = (props: Props) => {
+  const {parentId, onRecurrenceRuleUpdated, recurrenceRule} = props
+  const [recurrenceInterval, setRecurrenceInterval] = React.useState(
+    recurrenceRule ? recurrenceRule.options.interval : 1
+  )
+  const [recurrenceDays, setRecurrenceDays] = React.useState<Day[]>(
+    recurrenceRule
+      ? recurrenceRule.options.byweekday.map(
+          (weekday) => ALL_DAYS.find((day) => day.intVal === weekday)!
+        )
+      : []
+  )
+  const [recurrenceStartTime, setRecurrenceStartTime] = React.useState<Date>(
+    recurrenceRule
+      ? recurrenceRule.options.dtstart
+      : dayjs()
+          .add(1, 'day')
+          .set('hour', 6)
+          .set('minute', 0)
+          .set('second', 0)
+          .set('millisecond', 0)
+          .toDate() // suggest 6:00 AM tomorrow
+  )
+
+  const {timeZone} = Intl.DateTimeFormat().resolvedOptions()
+  const {menuPortal, togglePortal, menuProps, originRef} = useMenu<HTMLDivElement>(
+    MenuPosition.LOWER_LEFT,
+    {
+      id: 'recurrenceStartTimePicker',
+      parentId,
+      isDropdown: true
+    }
+  )
+
+  const handleIntervalChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    try {
+      const interval = parseInt(e.target.value)
+      setRecurrenceInterval(interval)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const handleDayChange = (day: Day) => {
+    if (recurrenceDays.includes(day)) {
+      setRecurrenceDays(recurrenceDays.filter((d) => d !== day))
+    } else {
+      setRecurrenceDays([...recurrenceDays, day])
+    }
+  }
+
+  useEffect(() => {
+    const rrule =
+      recurrenceDays.length > 0
+        ? new RRule({
+            freq: Frequency.WEEKLY,
+            interval: recurrenceInterval,
+            byweekday: recurrenceDays.map((day) => day.rruleVal),
+            dtstart: dayjs(recurrenceStartTime).utc().toDate(),
+            //TODO: this causes rrule to provide 'Invalid Date' for the next occurrences - see https://github.com/jakubroztocil/rrule/pull/547
+            tzid: timeZone
```

Comment:
Seems like the `RRule` library isn't actively maintained anymore. I propose to fork and maintain our own version of it. @mattkrick @Dschoordsch do you have any concerns or thoughts about that? this issue is the only blocker of this PR.

Reply:
There was a [comment yesterday](https://github.com/jakubroztocil/rrule/pull/547#issuecomment-1421216856) from what seems to be one of the maintainers, so I wouldn't consider this unmaintained. I see no issues having our own fork with this fix if we need it right now. Maintaining a date handling library long term though sounds quite frightening to me.

Reply:
yeah, will wait a little bit longer and see what happens.